includes/polling/mempools/hrstorage.inc.php

```diff
@@ -12,5 +14,9 @@
 
 $mempool['units'] = $entry['hrStorageAllocationUnits'];
 $mempool['used']  = ($entry['hrStorageUsed'] * $mempool['units']);
-$mempool['total'] = ($entry['hrStorageSize'] * $mempool['units']);
+if ($device['sysObjectID'] == '.1.3.6.1.4.1.12325.1.1.2.1.1') { // bsnmpd based devices, like FreeBSD, opnsense, pfsense ...
```

Comment:
Maybe just check if hrMemorySize is good or hrStorageSize is bad instead of if this is *BSD?

Reply:
I have no idea how to evaluate if hrMemorySize is good or hrStorageSize is bad, both values are present and can be polled.
But I know that on *BSD the hrStorageSize values is wrong.

Reply:
So there are values there, they are just wrong?? That seems really odd.  What are the two values?

Reply:
here can you read some explanation to this behavior:
https://lists.freebsd.org/pipermail/freebsd-stable/2014-March/078146.html
So that is a old one, but still the same bug on bsnmpd

Reply:
in #7541 this issue was addressed too, but no one had the right solution :-)